font_rename.py

```diff
@@ -79,18 +79,37 @@ def rename_font(filepath: Path):
 
 
 def unpack_ttc(filepath: Path):
-    collection = TTCollection(str(filepath.resolve()))
+    try:
+        collection = TTCollection(str(filepath.resolve()))
+    except:
+        print(f"Failed to parse {filepath}, ignore")
+        return
     for font in collection.fonts:
         ttf_path = filepath.parent / f"{get_font_name(font)}.ttf"
         font.save(ttf_path)
         print(f"{filepath} -> {ttf_path}")
     filepath.unlink()
 
 
+def unpack_otc(filepath: Path):
+    try:
+        collection = TTCollection(str(filepath.resolve()))
+    except:
+        print(f"Failed to parse {filepath}, ignore")
+        return
+    for font in collection.fonts:
+        ttf_path = filepath.parent / f"{get_font_name(font)}.otf"
+        font.save(ttf_path)
+        print(f"{filepath} -> {ttf_path}")
+    filepath.unlink()
+
+
 def handle_file(filepath: Path):
     suffix = filepath.suffix.lower()
     if suffix == ".ttc":
         unpack_ttc(filepath)
+    if suffix == ".otc":
+        unpack_otc(filepath)
```

Comment:
unpack_otc is identical to unpack_ttc.
I don't have a strong background in fonts, but according to [Noto CJK's document](https://www.google.com/get/noto/help/cjk/), `.ttc` is the file extension of OTC files.
<img width="474" alt="图片" src="https://user-images.githubusercontent.com/876694/112155326-27d45100-8c20-11eb-84bc-b862ecf8a275.png">

So we can remove the `unpack_otc` function :
```python
    if suffix in (".ttc", ".otc"):
        unpack_ttc(filepath)
```

( I personally can't find any font ends with `.otc`. Can you give me an example? Thanks)

Reply:
I previously had two .otc files from somewhere on my computer, but they failed to load so I deleted them. For testing I created a .otc file using otf2otc from afdko. (unpack_otc is different from unpack_ttc, as it will save the font files as an .otf file)

Reply:
After doing further testing, I have noticed that the unpack functions always cause a file to end in .ttf, even if it is  a opentype font file. I am working on modifying the script to cause the files to be unpacked to their correct filename.